assets/aml-benchmark/components/dataset-preprocessor/spec.yaml

```diff
@@ -0,0 +1,56 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+type: command
+
+name:  dataset_preprocessor 
+display_name: Dataset Preprocessor
+description: Dataset Preprocessor
+version: 0.0.1
+is_deterministic: true
+
+inputs:
+  dataset: 
+    type: uri_file
+    description: >-
+      Path to load the dataset.
+    optional: False
+  template_input:
+    type: string
+    description: >-
+      A json dictionary where key is the name of the column enclosed in " " and associated dict value is presented using jinja template logic 
+      which will be used to extract respective value from the dataset. 
+      Example format:
+      {"<user_column_name>": {{key in the json file for this column}}, ....}. The processed output will be dumped to a 
+      jsonl file in this format: {"<user_column_name>": "<key in the json file for this column>", ....} 
+    optional: True
+  script_path:
+    type: uri_file
+    description: >-
+      Path to the custom preprocessor python script provided by user. If this is provided, then, this takes precedence.  
```

Comment:
1. Can we specify, if we provide this, what does it take precedence over?
2. Why not throw an error if conflicting parameters are set together?

Reply:
1. I will specify that it take precedence over template_input if both parameters (template_input and script_path) are provided.
2. I would prefer to resolve such things in the backend rather instead of throwing an error. Let's discuss on this.

Reply:
> I would prefer to resolve such things in the backend rather instead of throwing an error.

We are trying to resolve an ambiguous scenario under an assumption. In such scenarios, we should make it clear to the user we support either A or B and if they are expecting anything different, they need to fix this. One can easily assume that A and B is also supported by running custom script first and then on output, running the template_input. This should be made clear to the user that it is not.

Reply:
@arun-rajora I have mentioned this in the component description. Although I have raised an error if both template_input and script_path is not provided.